controllers/issuer_controller.go

```diff
@@ -69,51 +76,58 @@ type IssuerReconciler struct {
 	PostSetupWithManager func(context.Context, schema.GroupVersionKind, ctrl.Manager, controller.Controller) error
 }
 
-func (r *IssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, returnedError error) {
+func (r *IssuerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	logger := log.FromContext(ctx).WithName("Reconcile")
 
 	logger.V(2).Info("Starting reconcile loop", "name", req.Name, "namespace", req.Namespace)
 
-	result, issuerStatusPatch, returnedError := r.reconcileStatusPatch(logger, ctx, req)
-	logger.V(2).Info("Got StatusPatch result", "result", result, "patch", issuerStatusPatch, "error", returnedError)
-	if issuerStatusPatch != nil {
-		cr, patch, err := ssaclient.GenerateIssuerStatusPatch(r.ForObject, req.Name, req.Namespace, issuerStatusPatch)
-		if err != nil {
-			return ctrl.Result{}, utilerrors.NewAggregate([]error{err, returnedError})
-		}
+	result, issuerStatusPatch, reconcileError := r.reconcileStatusPatch(logger, ctx, req)
+
+	if issuerStatusPatch == nil {
+		logger.V(2).Info("Got nil StatusPatch result", "result", result, "error", reconcileError)
+		return ctrl.Result{}, reconcileError
```

Comment:
**Question:** It seems like the error is already handled (since we are logging something). Why do return the error if it has already been handled?

Reply:
Ah, I think that this has become harder to understand, I'm going to revert https://github.com/cert-manager/issuer-lib/pull/42#discussion_r1314632848.
I think in general, we should re-check all the logging levels in this library. I think I log almost everything at level 2. But that is probably another PR.